[feature] domain redirects via Cloudfront

palasha opened the pull request:
There should be a canonical way to redirect from a naked domain like
http://foo.com or https://foo.com to https://www.foo.com, while
including the appropriate security headers like HSTS.

This is a first pass at such an abstraction.

I separated out the security-headers lambda since it is independently
useful in other cloudfront distribution settings. It's not very
configurable yet, and doesn't quite live up to the promise of its name,
but perhaps will get there in time.

Test Plan:
push the PR so I can try using it in another repo.

palasha commented:
ipv4 -> ipv6 remove duplicate resource name

:mag_right: [Click to see changes since last update](https://github.com/chanzuckerberg/cztack/compare/3dd8f4a4d27b13c1143e87317b8a9fb7fbea68d8..4d5f7a6b165e2e6971b5e6849a4285d531bc2dde).

palasha commented:
add missing var. prefix

:mag_right: [Click to see changes since last update](https://github.com/chanzuckerberg/cztack/compare/f1ecaae3d37c604e509c242dbab671baeddb05b4..384f81a5001a712232d1e3dc997cd07e701f3996).

palasha commented:
add missing lambda arn output for connecting with cloudfront

:mag_right: [Click to see changes since last update](https://github.com/chanzuckerberg/cztack/compare/846ae551c2fbee6901dd4e95234a118a6a6b2fb6..4add0ed10eb73fd7a038476ce26ac58c395623ac).

palasha commented:
move viewer_protocol_policy decalaration to correct block

:mag_right: [Click to see changes since last update](https://github.com/chanzuckerberg/cztack/compare/afc4172ba858d6647ee75e598bf5cbda34bc0a79..163f6798a7039ec4154971c81dbf32c9aed1e617).

palasha commented:
evaluate_target_health was a required attribute

:mag_right: [Click to see changes since last update](https://github.com/chanzuckerberg/cztack/compare/1768bbfa5753f1d4175bdc28b37d76f28a6e1f77..40d20e7952c7482ab555b191cf007117705ae8e8).

palasha commented:
replace . in lambda names with - to deal with domain named fogg components

:mag_right: [Click to see changes since last update](https://github.com/chanzuckerberg/cztack/compare/517039f69b380efd55b3540b9f57bd9715e5e414..ec1faf1c238def6111bfd4e3a76aa3f1649d20af).

palasha commented:
publish the security lambda and include qualified_arn in CF distro

:mag_right: [Click to see changes since last update](https://github.com/chanzuckerberg/cztack/compare/2b338e599774f92b4f28ed977a7f06eb9719a792..7ba56765f2401062fe26bc601920046aa7a739c7).

palasha commented:
add readmes and relative module references

:mag_right: [Click to see changes since last update](https://github.com/chanzuckerberg/cztack/compare/91e4b78d1381b58abef845908e618c1278186ebf..db38d05583b21ec3a67ae1920aaf75eb7fd59da8).

cziie commented on aws-lambda-edge-add-security-headers/outputs.tf:
```diff
@@ -0,0 +1,3 @@
+output qualified_arn {
```
📝 **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>`qualified_arn` output has no description

cziie commented on aws-lambda-edge-add-security-headers/main.tf:
```diff
@@ -0,0 +1,33 @@
+data archive_file lambda {
```
⚠️ **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing version constraint for provider "archive" in "required_providers"

ryanking reviewed (commented):
Looks good other than the lint comments (from cziie user).

palasha commented:
- pin `archive` provider version
- add output description for `qualified_arn`

:mag_right: [Click to see changes since last update](https://github.com/chanzuckerberg/cztack/compare/57ac9187d889cd19c112bb2a869f6f1cfa3bb706..c818a6153ebe2dfd7d63eaaec8e0e7358bf5d97d).

ryanking reviewed (changes requested):
really close!

palasha commented:
Rename module to be more generic

damien reviewed (approved):
Well done! This looks great!

![LGTM](https://media.giphy.com/media/oirLISmToyoeI/giphy.gif)

ryanking commented on aws-cloudfront-domain-redirect/module_test.go:
```diff
@@ -0,0 +1,18 @@
+func TestModule(t *testing.T) {
+	t.Skip("remove this for real tests")
+	test := tftest.Test{
+		Setup:    func(t *testing.T) *terraform.Options { return nil },
```
I would like to see a test here. I know that you aren't set up to run tests in the cztack accounts, so I will take it on.

ryanking reviewed (changes requested):
requesting changes, but I am doing to do them (write a test).
